File: php/chainofresponsibility/Director.php
<?php

/**
 * 主任
 */
class Director extends Leader
{
    public function handleRequest($request)
    {
        if ($request->getLeaveDays() < 3) {
            echo '员工：' . $request->getEmpName() . '，请假天数：' . $request->getLeaveDays() . '，理由：' .
                $request->getReason(), PHP_EOL;

            echo '主任：' . $this->name . '审批通过', PHP_EOL;
        } else {
            if ($this->nextLeader != null) {
                $this->nextLeader->handleRequest($request);
            }
        }
    }
}

File: java/pattern/src/chainofresponsibility/Director.java
package chainofresponsibility;

/**
 * 主任类
 */
public class Director extends Leader {
    public Director(String name) {
        super(name);
    }

    @Override
    public void handleRequet(LeaveRequest request) {
        if (request.getLeaveDays() < 3) {
            System.out.println("员工：" + request.getEmpName() + "，请假天数：" + request.getLeaveDays() + "，理由：" + request.getReason());
            System.out.println("主任：" + this.name + "，审批通过");
        } else {
            if (this.nextLeader != null) {
                this.nextLeader.handleRequet(request);
            }
        }
    }
}
